pypeit/core/datacube.py

```diff
@@ -267,6 +270,306 @@ def dar_correction(wave_arr, coord, obstime, location, pressure, temperature, re
+        theta (float):
+            The orientation angle of the 2D Gaussian
+        offset (float):
+            Constant offset
+
+    Returns:
+        gtwod (`numpy.ndarray`_): The 2D Gaussian evaluated at the coordinate (x, y)
+    """
+    # Extract the (x, y, z) coordinates of each pixel from the tuple
+    (x, y) = tup
+    # Ensure these are floating point
+    xo = float(xo)
+    yo = float(yo)
+    # Account for a rotated 2D Gaussian
+    a = (np.cos(theta)**2)/(2*sigma_x**2) + (np.sin(theta)**2)/(2*sigma_y**2)
+    b = -(np.sin(2*theta))/(4*sigma_x**2) + (np.sin(2*theta))/(4*sigma_y**2)
+    c = (np.sin(theta)**2)/(2*sigma_x**2) + (np.cos(theta)**2)/(2*sigma_y**2)
+    # Normalise so that the integrated flux is a parameter, instead of the amplitude
+    norm = 1/(2*np.pi*np.sqrt(a*c-b*b))
+    gtwod = offset + norm*intflux*np.exp(-(a*((x-xo)**2) + 2*b*(x-xo)*(y-yo) + c*((y-yo)**2)))
+    return gtwod.ravel()
+
+
+def rebinND(img, shape):
+    """
+    Rebin a 2D image to a smaller shape. For example, if img.shape=(100,100),
+    then shape=(10,10) would average the first 10x10 pixels into a single output
+    pixel, then next 10x10 pixels will be output into the next pixel
+
+    Args:
+        img (`numpy.ndarray`_):
+            A 2D input image
+        shape (:obj:`tuple`):
+            The desired shape to be returned. The elements of img.shape
+            should be an integer multiple of the elements of shape.
+
+    Returns:
+        img_out (`numpy.ndarray`_): The input image rebinned to shape
+    """
+    # Convert input 2D image into a 4D array to make the rebinning easier
+    sh = shape[0], img.shape[0]//shape[0], shape[1], img.shape[1]//shape[1]
+    # Rebin to the 4D array and then average over the second and last elements.
+    img_out = img.reshape(sh).mean(-1).mean(1)
+    return img_out
+
+
+def extract_standard_spec(stdcube, subsample=20, method='boxcar'):
+    """ Extract a spectrum of a standard star from a datacube
+
+    Args:
+        std_cube (`astropy.io.fits.HDUList`_):
+            An HDU list of fits files
+        subsample (int):
+            Number of pixels to subpixellate spectrum when creating mask
+        method (str):
+            Method used to extract standard star spectrum
+
+    Returns:
+        wave (`numpy.ndarray`_): Wavelength of the star.
+        Nlam_star (`numpy.ndarray`_): counts/second/Angstrom
+        Nlam_ivar_star (`numpy.ndarray`_): inverse variance of Nlam_star
+        gpm_star (`numpy.ndarray`_): good pixel mask for Nlam_star
+    """
+    # Extract some information from the HDU list
+    flxcube = stdcube['FLUX'].data.T.copy()
+    varcube = stdcube['VARIANCE'].data.T.copy()
+    numwave = flxcube.shape[2]
+
+    # Setup the WCS
+    stdwcs = wcs.WCS(stdcube['FLUX'].header)
+    wcs_wav = stdwcs.wcs_pix2world(np.vstack((np.zeros(numwave), np.zeros(numwave), np.arange(numwave))).T, 0)
+    wave = wcs_wav[:, 2] * 1.0E10 * units.AA
+
+    # Generate a whitelight image
```

Comment:
while this is a pretty simple 4 lines of code, I wonder whether it should be a method somewhere.  Just consider it